src/features/hotspots/details/HotspotStatusBanner.tsx

```diff
@@ -5,40 +5,63 @@ import React, { useState, useEffect, memo, useMemo, useCallback } from 'react'
 import { useTranslation } from 'react-i18next'
 import { useSelector } from 'react-redux'
 import Close from '@assets/images/close.svg'
+import { formatDistanceToNow } from 'date-fns'
 import Box from '../../../components/Box'
 import Text from '../../../components/Text'
 import { RootState } from '../../../store/rootReducer'
 import { Theme } from '../../../theme/theme'
 import { useColors } from '../../../theme/themeHooks'
 import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
 import animateTransition from '../../../utils/animateTransition'
+import shortLocale from '../../../utils/formatDistance'
 import usePrevious from '../../../utils/usePrevious'
+import { HotspotSyncStatus } from '../../../utils/hotspotUtils'
 
 type Props = BoxProps<Theme> & {
   hotspot: Hotspot | Witness
   visible: boolean
   onDismiss: () => void
+  syncStatus: HotspotSyncStatus | null
 }
 const HotspotStatusBanner = ({
   hotspot,
   visible: propsVisible,
   onDismiss,
+  syncStatus,
   ...boxProps
 }: Props) => {
   const blockHeight = useSelector(
     (state: RootState) => state.heliumData.blockHeight,
   )
-  const syncStatuses = useSelector(
-    (state: RootState) => state.hotspots.syncStatuses,
-  )
   const { t } = useTranslation()
   const { orangeExtraDark } = useColors()
   const [visible, setVisible] = useState(false)
   const prevHotspotAddress = usePrevious(hotspot.address)
-  const hotspotSyncStatus = useMemo(() => syncStatuses[hotspot.address], [
-    hotspot.address,
-    syncStatuses,
-  ])
+
+  const syncStatusMessage = useMemo(() => {
+    if (!blockHeight || !syncStatus || hotspot.status?.online !== 'online')
+      return null
+
+    const timeAgo = hotspot.status?.timestamp
```

Comment:
I'm seeing this logic repeat, can it be centralized somewhere? like a `<SyncStatusMessage />`?